### PR TITLE
Desktop profile fixes

### PR DESCRIPTION
**Pull request by davide125**
- gate x86 firmware to the architecture
- move KDE-specific packages to the KDE profile

### CODE REVIEW

File: mkosi.profiles/desktop/mkosi.conf
Comment (davide125) [inline mkosi.profiles/desktop/mkosi.conf:7]: [KDb](https://community.kde.org/KDb) seems an odd thing to install by default. Did you mean [kbd](http://www.kbd-project.org/) ?

**Comment by davide125** [comment]
Refactored a few more Arch and Fedora differences into their own configs, and added `/etc/dbus-1` to tmpfiles as otherwise a bunch of services fail on boot.

File: mkosi.profiles/desktop/mkosi.conf
Comment (daandemeyer) [reply mkosi.profiles/desktop/mkosi.conf:7]: oops, good catch, can you switch this to `kbd` and keep it here?

File: mkosi.profiles/desktop/mkosi.extra/usr/lib/tmpfiles.d/10-particleos-desktop.conf
Comment (daandemeyer) [inline mkosi.profiles/desktop/mkosi.extra/usr/lib/tmpfiles.d/10-particleos-desktop.conf]: Isn't this something to fix in Fedora? dbus should looks in /usr as well no?

File: mkosi.profiles/desktop/mkosi.conf.d/fedora/mkosi.conf.d/mkosi.conf
Comment (daandemeyer) [inline mkosi.profiles/desktop/mkosi.conf.d/fedora/mkosi.conf.d/mkosi.conf]: This condition can be dropped, already handled by `fedora/mkosi.conf`

File: mkosi.profiles/desktop/mkosi.conf.d/fedora/mkosi.conf.d/mkosi.conf
Comment (daandemeyer) [inline mkosi.profiles/desktop/mkosi.conf.d/fedora/mkosi.conf.d/mkosi.conf]: Might as well drop this until we actually decide to support 32-bit (which I guess we never will)

File: mkosi.profiles/desktop/mkosi.conf.d/arch/mkosi.conf.d/mkosi.conf
Comment (daandemeyer) [inline mkosi.profiles/desktop/mkosi.conf.d/arch/mkosi.conf.d/mkosi.conf]: This can be dropped as the condition is handled by `arch/mkosi.conf` already

File: mkosi.profiles/desktop/mkosi.conf.d/arch/mkosi.conf.d/mkosi.conf
Comment (daandemeyer) [inline mkosi.profiles/desktop/mkosi.conf.d/arch/mkosi.conf.d/mkosi.conf]: We don't support 32-bit and probably never will so I'd drop this.

File: mkosi.profiles/desktop/mkosi.conf.d/fedora/mkosi.conf
Comment (daandemeyer) [inline mkosi.profiles/desktop/mkosi.conf.d/fedora/mkosi.conf]: Didn't we just switch to `tuned` in Fedora 41?

File: mkosi.profiles/kde/mkosi.conf
Comment (daandemeyer) [inline mkosi.profiles/kde/mkosi.conf]: Curious why you think flatpak would be KDE specific? It works on GNOME as well no?

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]

File: mkosi.profiles/desktop/mkosi.extra/usr/lib/tmpfiles.d/10-particleos-desktop.conf
Comment (davide125) [reply mkosi.profiles/desktop/mkosi.extra/usr/lib/tmpfiles.d/10-particleos-desktop.conf]: So I added this because otherwise `rtkit-daemon` (pulled in by pipewire) fails to start on boot, because `/etc/dbus-1/system.d/org.freedesktop.RealtimeKit1.conf`. I suppose we should fix that rpm to put that in `/usr` instead?

File: mkosi.profiles/desktop/mkosi.conf.d/fedora/mkosi.conf
Comment (davide125) [reply mkosi.profiles/desktop/mkosi.conf.d/fedora/mkosi.conf]: Uh that's a good point, I didn't see `power-profiles-daemon` on any of my Fedora systems, which is why I moved it under arch. I'll double check on tuned. btw, at least on Fedora tuned won't start unless `/etc/tuned` is present, so we might need to add that too to tmpfiles (or fix it upstream).

File: mkosi.profiles/kde/mkosi.conf
Comment (davide125) [reply mkosi.profiles/kde/mkosi.conf]: It's not KDE specific, but it seemed to me something more usecase specific than something we should have in a base profile (like, if you're making an appliance-style system you probably wouldn't want it). Debatable of course, like most things :)

File: mkosi.profiles/desktop/mkosi.extra/usr/lib/tmpfiles.d/10-particleos-desktop.conf
Comment (davide125) [reply mkosi.profiles/desktop/mkosi.extra/usr/lib/tmpfiles.d/10-particleos-desktop.conf]: Yeah, this is explicitly installed as a config file: https://src.fedoraproject.org/rpms/rtkit/blob/rawhide/f/rtkit.spec#_77

Comment (daandemeyer) [reply mkosi.profiles/desktop/mkosi.extra/usr/lib/tmpfiles.d/10-particleos-desktop.conf]: @davide125 Can you file a PR to move it? I think @keszybz maintains rtkit in Fedora

File: mkosi.profiles/kde/mkosi.conf
Comment (daandemeyer) [reply mkosi.profiles/kde/mkosi.conf]: Good point, it doesn't really fit in with the rest of the packages in the desktop profile. We should probably split a graphics profile from the desktop profile with all the graphics stuff and then the desktop profile can have flatpak in it. For now feel free to drop flatpak, I can just add it to my local config.

File: mkosi.profiles/desktop/mkosi.extra/usr/lib/tmpfiles.d/10-particleos-desktop.conf
Comment (davide125) [reply mkosi.profiles/desktop/mkosi.extra/usr/lib/tmpfiles.d/10-particleos-desktop.conf]: https://src.fedoraproject.org/rpms/rtkit/pull-request/3

File: mkosi.profiles/desktop/mkosi.conf.d/fedora/mkosi.conf
Comment (davide125) [reply mkosi.profiles/desktop/mkosi.conf.d/fedora/mkosi.conf]: Ok, looks like on Fedora we should use `tuned-ppd` instead of `power-profiles-daemon`: https://fedoraproject.org/wiki/Changes/TunedAsTheDefaultPowerProfileManagementDaemon

File: mkosi.profiles/kde/mkosi.conf
Comment (davide125) [reply mkosi.profiles/kde/mkosi.conf]: Dropped

**Comment by DaanDeMeyer** [comment]
I just opened https://github.com/systemd/systemd/pull/35013 in systemd which will allow us to keep all tmpfiles lines in the same snippet in the main config so let's wait until that is merged and then all the tmpfiles stuff can be added to the existing file instead of adding new files.

**Comment by DaanDeMeyer** [comment]
@davide125 L? is now supported by systemd, I also added instructions on how to build systemd distribution packages from source and how to get ParticleOS to use them. Can you merge the tmpfiles snippets into the main one using `L?`?

File: mkosi.profiles/desktop/mkosi.extra/usr/lib/tmpfiles.d/10-particleos-desktop.conf
Comment (daandemeyer) [reply mkosi.profiles/desktop/mkosi.extra/usr/lib/tmpfiles.d/10-particleos-desktop.conf]: I think we can drop this now

**Comment by davide125** [comment]
Updated, thanks!

**Pull request merged**
